cdap-support-bundle/bin/observability.py

```diff
@@ -0,0 +1,494 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © 2014-2023 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# conf.py to build high-level document comprised of multiple manuals
+# Relies on settings in common_conf.py
+
+# Search Index
+# Includes handler to build a common search index of all manuals combined
+
+import base64
+import json
+import os
+import shutil
+import subprocess
+import sys
+from datetime import datetime
+
+import click
+
+
+@click.command()
+@click.option(
+    "--namespace",
+    help="Mandatory parameter. Kubernetes namespace where CDAP is installed",
+    type=str,
+)
+@click.option(
+    "--instancename",
+    help="Mandatory parameter. CDAP instance name",
+    type=str,
+)
+@click.option(
+    "--username",
+    help="Mandatory parameter. CDAP User Name",
+    type=str,
+)
+@click.option(
+    "--password",
+    help="Mandatory parameter. CDAP Password",
+    type=str,
+)
+def main(namespace, instancename, username, password):
+    """Log collector program for CDAP."""
+
+    # if namespace provided as parameter then assign else take input from user
+    if namespace is None or instancename is None or username is None or password is None:
+        print('Missing mandatory input parameters. Please visit help using --help')
+        exit()
+
+
+    passtoken = base64.b64encode(bytes(f'{username}:{password}', 'utf-8'))
+    passtoken = passtoken.decode('utf-8')
+
+    # running command to get AUTH_TOKEN
+    output = subprocess.getoutput(
+        f'kubectl -n {namespace} exec -t cdap-{instancename}-appfabric-0 -c appfabric -- bash -c \'curl -s cdap-{instancename}-external-authentication:10009/token -H "Authorization:Basic {passtoken}"\'')
+    try:
+        str_obj = json.loads(output)
+    except:
+        print('Invalid input parameters. Please check if you have provided correct values for all input parameters.')
+        exit()
+    auth_token = str_obj['access_token']
+
+    # creating object for current datetime
+    update_progress(1)
```

Comment:
How do we decide to update progress %? 
Does it take a long time to generate (10s of seconds) the tarball? If not, the progress bar might not be necessary

Reply:
in case of getting app-fabric logs it can take more than 2 min